### PR TITLE
Add offscreen rendering example/docs

### PR DESCRIPTION
**Pull request by violinyanev**
Kept very simple, just single target/pass with color+depth buffer. Result should be quite straightforward and easy to
understand and demonstrate the basic features of composer for offscreen rendering.

### CODE REVIEW

File: offscreen/manual.md [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+This Source Code Form is subject to the terms of the Mozilla Public License, v. 2.0.
+If a copy of the MPL was not distributed with this file, You can obtain one at http://mozilla.org/MPL/2.0/.
+-->
+
+# Offscreen rendering example
+
+![](./docs/viewport_preview.png)
+
+This example shows how to perform offscreen rendering with the Ramses Composer (also called render-to-texture).
+This article explains the following concepts:
```
Comment (bojackHaasman) [inline offscreen/manual.md:16]: The new line is not applied:
image

File: offscreen/manual.md [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+times - this technique is a requirement for offscreen rendering.
+
+Let's have a look at the contents of the project.
+
+## Scene graph and resources
+
+After opening the project, we notice a few more things compared to the [monkey example](../monkey/manual.md).
+
+To keep this example simple, we render a single monkey head without a script.
+In the Scene View, there are two cameras - one for rendering the monkey, one for rendering the quad:
```
Comment (bojackHaasman) [inline offscreen/manual.md:35]:
```suggestion
In the Scene View, there are two cameras - one for rendering the monkey and one for rendering the quad:
```

File: offscreen/manual.md [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+
+![](./docs/resources.png)
+
+These new objects define how the offscreen rendering should work, i.e. which objects are
+rendered to which buffer at which time.
+
+Let's start with the RenderLayers. We need two such layers - one for the monkey, one for the quad.
+Each of them uses a different tag to filter their content - `monkey` and `quad` respectively.
+
+Here is how the `RenderLayer` of the monkey looks like this (it uses a `monkey` tag to collect the monkey mesh):
```
Comment (bojackHaasman) [inline offscreen/manual.md:49]:
```suggestion
Here is how the `RenderLayer` of the monkey looks like (it uses a `monkey` tag to collect the monkey mesh):
```

File: offscreen/manual.md [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+is a collection of one or multiple buffers for rendering.
+Those can be color buffers, and optionally a depth and/or stencil buffer. These buffers must all have the same size, and must
+fulfill other criteria too - e.g. you can have at most one depth buffer, and they can't exceed a given size. The Ramses Composer
+will report errors in the configuration of the `RenderTarget` if such are found. Note that the FBO configuration is a hardware
+detail - not all formats work on all hardware devices. Some devices have a limit on e.g. how many floating point textures can
+be created, and not all devices support 16 or 32 bit float channels. Be careful which formats you really need and cross-check
+with your target hardware!
+
+In this example, we have two such buffers - one for color, and one for depth information. Note that they have
+a size which is different than the framebuffer, and have a format (RGBA8 and Depth24 respectively):
```
Comment (bojackHaasman) [inline offscreen/manual.md:79]: what is the framebuffer size in this example, or how can I tell that it is different?

File: offscreen/manual.md [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+
+This example is created in a way similar to the [monkey example](../monkey/manual.md#recreate-sample).
+We only highlight the steps specific to this example here:
+
+* Create a new project and delete the default content
+* Create a standard "Suzanne" monkey (same as in [monkey example](../monkey/manual.md))
+    * You can also just copy the project and delete two of the monkey heads
+* Create the Quad
+    * Create a standard "Plane" mesh in Blender and export as glTF (quad.gltf)
+        * Make sure to export the exture coordinates!
```
Comment (bojackHaasman) [inline offscreen/manual.md:114]:
```suggestion
        * Make sure to export the texture coordinates!
```

**Review by bojackHaasman — CHANGES_REQUESTED** [review]

File: offscreen/manual.md (hunk 1 above)
Comment (violinyanev) [reply offscreen/manual.md:16]: It's not supposed to. 2 lines in markdown means newline in the rendered version. This \n is just to keep the line a bit shorter

File: offscreen/manual.md (hunk 4 above)
Comment (bojackHaasman) [reply offscreen/manual.md:79]: yes, better now

File: offscreen/manual.md (hunk 1 above)
Comment (bojackHaasman) [reply offscreen/manual.md:16]: Ok, still it's a bit crooked: This example shows  ... next sentence ... this article explains

File: offscreen/manual.md [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+This Source Code Form is subject to the terms of the Mozilla Public License, v. 2.0.
+If a copy of the MPL was not distributed with this file, You can obtain one at http://mozilla.org/MPL/2.0/.
+-->
+
+# Offscreen rendering example
+
+![](./docs/viewport_preview.png)
+
+This example shows how to perform offscreen rendering with the Ramses Composer (also called render-to-texture).
+This article explains the following concepts:
```
Comment (bojackHaasman) [inline offscreen/manual.md:16]:
```suggestion
Following concepts will be explained:
```

**Review by bojackHaasman — CHANGES_REQUESTED** [review]

**Review by bojackHaasman — APPROVED** [review]

**Pull request merged**